### PR TITLE
Fix conf-ssl for OSX/homebrew.

### PR DESCRIPTION
**Pull request by toots**
This should take care of tezos/tezos#123 and savonet/ocaml-ssl#33

### CODE REVIEW

**Comment by gasche** [comment]
I understand that this fix makes `conf-ssl`'s install more successful, but is it correct with respect to the assumptions of the packagers that depend on `conf-ssl`? If they assumed that `pkg-config openssl` works, then their build/configure script may be broken by this change. I don't know about SSL-related packages, maybe that never happens.

It seems to me that accepting this patch is the better thing to do (it may not be perfect but it cannot hurt and does improve some cases users care about). But I suppose that a longer-term solution (in a more expressive opam2 world) would need to flow more information towards consumers of `conf-ssl` (by defining package variables, etc.) on how to access the dependency -- if we don't only support the systems on which the access is standard.

**Comment by toots** [comment]
First, thanks for pushing back on this! I had to look at some it and found an important issue on `ocaml-ssl`. This is fixed with the release of `ssl.0.5.5` which is included in this PR. Now, baring this issue, here's the full story:

The problem on `OSX` comes from the fact that the system ships their own SSL implementation and no header (`ssl.h`) for it.

Thus, `homebrew` ships a `openssl` package that is build only and is used as a dependency for software relying on the original `openssl`. Therefore, its headers and libraries are located in a directory that is intentionally not in any of the usual paths searched for those:
```
% brew info openssl
openssl: stable 1.0.2l (bottled) [keg-only]
SSL/TLS cryptography library
https://openssl.org/
(...)
==> Caveats
(...)
This formula is keg-only, which means it was not symlinked into /usr/local,
because Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries.

If you need to have this software first in your PATH run:
  echo 'export PATH="/usr/local/opt/openssl/bin:$PATH"' >> ~/.zshrc

For compilers to find this software you may need to set:
    LDFLAGS:  -L/usr/local/opt/openssl/lib
    CPPFLAGS: -I/usr/local/opt/openssl/include
For pkg-config to find this software you may need to set:
    PKG_CONFIG_PATH: /usr/local/opt/openssl/lib/pkgconfig
```

Changing those environment variables permanently through shell config files isn't recommended because it will leak into other software that rely on the mainstream `OSX` ssl library.

`conf-openssl` declares `homebrew`'s `openssl` package as dependency so, it is only fair that it also points `pkg-config` to its _documented_ location, according to the `homebrew` info:
```
% cat conf-openssl/conf-openssl.1/opam
(...)
build: [
  ["pkg-config" "openssl"] {os != "darwin" & os != "freebsd" & os != "openbsd" & os != "netbsd"}
  ["sh" "-exec" "PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig pkg-config openssl"] {os = "darwin"}
  ] { os = "darwin" }
]
depends: ["conf-pkg-config"]
depexts: [
(...)
  [["osx" "homebrew"] ["openssl"]]
(...)
]
```

This is also the way that the `ssl` package is built on `OSX`:
```
% cat ssl/ssl.0.5.5/opam
(...)
build: [
  ["./configure" "--prefix" prefix] { os != "darwin" }
   "./configure --prefix %{prefix}% CPPFLAGS=\"-I/usr/local/opt/openssl/include\" LDFLAGS=\"-L/usr/local/opt/openssl/lib\""
  [make]
]
(...)
```

Finally, if you remember that `OCaml` modules keep track of link and build variables, this trickles down to the compiled modules:
```
% ocamlobjinfo /path/to/ssl.cmxa
File /path/to/ssl.cmxa
Extra C object files: -lssl_stubs -lssl -lcrypto
Extra C options: -L/usr/local/opt/openssl/lib
Name: Ssl
(...)
```

Which, eventually, is also picked up when linking against them, for instance:
```
% otool -L `which liquidsoap`
/path/to/liquidsoap:
(...)
	/usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
(...)
```

In conclusion, I'd argue that these flags are actually needed to make sure that we preserved API and ABI compatibility when building `ssl`-related modules and binaries using `opam` on `OSX`. The situation with system vs. `homebrew` `ssl` is most likely going to stay the same for a while and that custom `homebrew` is the actual, documented path and method to link against `openssl` as provided by the package manager.

Hope that this responds to your concerns.

**Comment by gasche** [comment]
Indeed, thanks for elaborating; merged.
